figures/compat.py

```diff
@@ -41,9 +43,13 @@ def course_grade(learner, course):
     Returns the course grade for the specified learner and course
     """
     if RELEASE_LINE == 'ginkgo':
-        return CourseGradeFactory().create(learner, course)
+        course_grade = CourseGradeFactory().create(learner, course)
     else:  # Assume Hawthorn or greater
-        return CourseGradeFactory().read(learner, course)
+        course_grade = CourseGradeFactory().read(learner, course)
+
+    persistent_course_grade = PersistentCourseGrade.objects.filter(user_id=learner.id, course_id=course.id).first()
```

Comment:
`PersistentCourseGrade` class provides a `read` function for this no? Something like
```
        persistent_grade = PersistentCourseGrade.read(user.id, course_data.course_key)
```

Reply:
@rehan99000 read function doesn't return passed_timestamp. For that, we would need to filter courses using model. CourseGradeFactiory also calls PersistentCourseGrade.read under the hood but it brings some additional data too.